pkg/deployments/src/network.ts

```diff
@@ -42,25 +42,53 @@ export function getContractDeploymentTransactionHash(deployedAddress: string, ne
   return txHash;
 }
 
-export function saveContractDeploymentAddresses(task: Task): void {
-  if (task.network === 'hardhat') return;
+/**
+ * Saves file with canonical deployment addresses for all the given tasks in the given network,
+ */
+export function saveContractDeploymentAddresses(tasks: Task[], network: string): void {
+  if (network === 'hardhat') return;
+
+  const allTaskEntries = buildContractDeploymentAddressesEntries(tasks);
+  const filePath = path.join(CONTRACT_ADDRESSES_DIRECTORY, `${network}.json`);
+
+  fs.writeFileSync(filePath, _stringifyEntries(allTaskEntries));
+}
+
+/**
+ * Builds object that maps deployment addresses to {task ID, contract name} for all the given tasks.
+ */
+export function buildContractDeploymentAddressesEntries(tasks: Task[]): object {
+  let allTaskEntries = {};
+
+  for (const task of tasks) {
+    const taskEntries = Object.fromEntries(
+      Object.entries(task.output({ ensure: false })).map(([name, address]) => [address, { task: task.id, name }])
+    );
+    allTaskEntries = {
+      ...allTaskEntries,
+      ...taskEntries,
+    };
+  }
+
+  return allTaskEntries;
+}
 
-  const newEntries = Object.fromEntries(
-    Object.entries(task.output({ ensure: false })).map(([name, address]) => [address, { task: task.id, name }])
-  );
+/**
+ * Returns true if the existing deployment addresses file stored in `CONTRACT_ADDRESSES_DIRECTORY` matches the
+ * canonical one for the given set of tasks in the given network; false otherwise.
```

Comment:
```suggestion
 * Returns true if the existing deployment addresses file stored in `CONTRACT_ADDRESSES_DIRECTORY` matches the
 * canonical one for the given network; false otherwise.
```